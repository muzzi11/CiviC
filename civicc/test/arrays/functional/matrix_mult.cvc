extern void printInt( int val);
extern void printFloat( float val);

extern int scanInt( );
extern float scanFloat( );

extern void printSpaces( int num);
extern void printNewlines( int num);


export int main()
{
	//Read dimensions of matrix from input
	int rowsMatrixA =5;
	int colMatrixA = 5;

	int rowsMatrixB =5;
	int colMatrixB = 5;

	int[rowsMatrixA, colMatrixA] matrixA;
	int[rowsMatrixB, colMatrixB] matrixB;
	int[rowsMatrixA, colMatrixB] matrixProduct;

	//Product cannot be done if the following condition does not stand
	if (colMatrixA == rowsMatrixB)
	{
		//Fill in Matrix A
		for (int i = 0, rowsMatrixA)
		{
			for (int j = 0, colMatrixA)
			{
				matrixA[i, j] = i + j;
				printInt(matrixA[i, j]);
				printSpaces(3);
			}
			printNewlines(1);
		}
        printNewlines(1);

		//Fill in Matrix B
		for (int i = 0, rowsMatrixB)
		{
			for (int j = 0, colMatrixB)
			{
				matrixB[i, j] = i + j;
				printInt(matrixB[i, j]);
				printSpaces(3);
			}
			printNewlines(1);
		}
        printNewlines(1);

		//Do Multiplication
		for (int i = 0, rowsMatrixA)
		{
			for (int j = 0, colMatrixB)
			{
				matrixProduct[i, j] = 0;
				for (int m = 0, colMatrixA)
				{
					matrixProduct[i, j] = matrixProduct[i, j] + (matrixA[i, m] * matrixB[m,j]);
				}
			}
		}


		for (int i = 0, rowsMatrixA)
		{
			for (int j = 0, colMatrixB)
			{
				printInt(matrixProduct[i, j]);
						printSpaces(3);
			}
			printNewlines(1);
		}
	}
	return 0;
}


