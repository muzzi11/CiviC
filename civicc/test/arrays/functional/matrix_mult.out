0   1   2   3   4
1   2   3   4   5
2   3   4   5   6
3   4   5   6   7
4   5   6   7   8

0   1   2   3   4
1   2   3   4   5
2   3   4   5   6
3   4   5   6   7
4   5   6   7   8

30   40   50   60   70
40   55   70   85   100
50   70   90   110   130
60   85   110   135   160
70   100   130   160   190
